File: CanzoniereWPF/Design/DesignDataService.vb
Imports System.Collections.ObjectModel
Imports Enzo.Music

Public Class DesignDataService
    Implements IDataService

    Public Async Function CreateIndexAsync(folderPath As String) As Task Implements IDataService.CreateIndexAsync
        Await Task.Run(Sub()
                           If 1 = 0 Then
                               ' DoNothing
                           End If
                       End Sub)
        Return
    End Function

    Public Function GetFileElementsFromFolderAsync(folderPath As String) As Task(Of List(Of FileElement)) Implements IDataService.GetFileElementsFromFolderAsync
        Dim ret As New List(Of FileElement) From {
            New FileElement() With {.FileName = "..", .FilePath = "c:\", .IsFolder = True, .IsPreviousFolder = True, .Title = ".."},
            New FileElement() With {.FileName = "Cartella d'esempio", .FilePath = "c:\temp\folder1", .IsFolder = True, .Title = "folder1"},
            New FileElement() With {.FileName = "kantoj1", .FilePath = "c:\temp\file1.kantoj", .IsListOfFiles = True, .Title = "file1.kantoj"},
            New FileElement() With {.FileName = "kanto1", .FilePath = "c:\temp\file1.kanto", .Title = "Alleluia (Toronto 2002)"},
            New FileElement() With {.FileName = "kanto2", .FilePath = "c:\temp\file2.kanto", .Title = "Altra canzone della lista con nome molto lungo"},
            New FileElement() With {.FileName = "kanto3", .FilePath = "c:\temp\file3.kanto", .Title = "Canzone con errori", .HasErrors = True},
            New FileElement() With {.FileName = "kanto4", .FilePath = "c:\temp\file4.kanto", .Title = "File mancante", .Exists = False}
        }
        Return Task.FromResult(ret)
    End Function

    Public Async Function GetCurrentFolderAsync() As Task(Of Uri) Implements IDataService.GetCurrentFolderAsync
        Return Await Task.FromResult(New Uri(IO.Directory.GetCurrentDirectory()))
    End Function

    Public Async Function GetCanzoneFromFilePathAsync(filePath As String) As Task(Of Canzone) Implements IDataService.GetCanzoneFromFilePathAsync
        Return Await Task.FromResult(New Canzone() With {.Titolo = "Canzone 1", .Autore = "Autore 1"})
    End Function
End Class

File: KantaroWpf/Services/DesignDataService.cs
using Enzo.Music.KantaroWpf.Services;
using Enzo.Music.KantaroWpf.Models;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Windows.Interop;

namespace Enzo.Music.KantaroWpf.Services
{
    class DesignDataService : IDataService
    {
        public async Task CreateIndexAsync(string folderPath)
        {
            await Task.Run(() => null);
        }

        public Task<Canzone?> GetCanzoneFromFilePathAsync(string filePath)
        {
            return Task.FromResult(new Canzone { Titolo = "Canzone 1", Autore = "Autore 1" }??null);
        }

        public Task<string> GetCurrentFolderAsync()
        {
            return Task.FromResult(System.IO.Directory.GetCurrentDirectory());
        }

        public Task<List<FileElement>> GetFileElementsFromFolderAsync(string folderPath)
        {
            List<FileElement> ret = new() {
                new FileElement {FileName = "..", FilePath = "c:\\", IsFolder = true, IsPreviousFolder = true, Title = ".."},
                new FileElement {FileName = "Cartella d'esempio", FilePath = "c:\\temp\\folder1", IsFolder = true, Title = "folder1"},
                new FileElement {FileName = "kantoj1", FilePath = "c:\\temp\\file1.kantoj", IsListOfFiles = true, Title = "file1.kantoj"},
                new FileElement {FileName = "kanto1", FilePath = "c:\\temp\\file1.kanto", Title = "Alleluia (Toronto 2002)"},
                new FileElement {FileName = "kanto2", FilePath = "c:\\temp\\file2.kanto", Title = "Altra canzone della lista con nome molto lungo"},
                new FileElement {FileName = "kanto3", FilePath = "c:\\temp\\file3.kanto", Title = "Canzone con errori", HasErrors = true},
                new FileElement {FileName = "kanto4", FilePath = "c:\\temp\\file4.kanto", Title = "File mancante", Exists = false}
                };
            return Task.FromResult(ret);
        }
    }
}
